### PR TITLE
refactor: Aggregate errors in SaveBlock using errors.Join

### LINKED ISSUE #1490: Unnecessary usage of external lib for multi-errors
<meta charset="utf-8"><b style="font-weight:normal;" id="docs-internal-guid-e69d3adc-7fff-f2e4-37cb-11840cf43b46"><h3 dir="ltr" style="line-height:1.38;margin-top:16pt;margin-bottom:4pt;"><span style="font-size:13.999999999999998pt;font-family:Roboto,sans-serif;color:#434343;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Finding 025 - Unnecessary usage of external lib for multi-errors</span></h3><div dir="ltr" style="margin-left:0pt;" align="left">
ID | 025
-- | --
Finding | Unnecessary usage of external lib for multi-errors
Severity | 1 - Informational
Description | The go.uber.org/multierr package is used to collect and chain errors in methods where multiple errors need to be handled as one. The Go stdlib allows for this functionality.
Recommendation | Consider using the Go errors.Join API over the go.uber.org/multierr package.
Code References | https://github.com/rollkit/rollkit/blob/eccdd0f1793a5ac532011ef4d896de9e0d8bcb9d/go.mod#L32

</div></b>

### PR DESCRIPTION
**Pull request by Chirag018**
Closes #1490

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

## Summary by CodeRabbit

- **Refactor**
  - Improved error handling in save operations for enhanced reliability.

### CODE REVIEW

**Comment by Manav-Aggarwal** [comment]
Thanks for submitting this PR @Chirag018! Looks great overall but seems like there are merge conflicts for the PR. Please feel free to address :)

File: store/store.go [hunk 1]
```diff
@@ -79,17 +79,18 @@ func (s *DefaultStore) SaveBlock(ctx context.Context, block *types.Block, commit
...
+
+	putErr = errors.Join(
+		putErr,
+		bb.Put(ctx, ds.NewKey(getBlockKey(hash)), blockBlob),
+		bb.Put(ctx, ds.NewKey(getCommitKey(hash)), commitBlob),
+		bb.Put(ctx, ds.NewKey(getIndexKey(block.Height())), hash[:]),
+	)
+
+	if putErr != nil {
+		bb.Discard(ctx)
```
Comment (Manav-Aggarwal) [inline store/store.go:92]: Note that `defer bb.Discard(ctx)` is called earlier so this is unnecessarly.

**Review by Manav-Aggarwal — CHANGES_REQUESTED** [review]

File: store/store.go (hunk 1 above)
Comment (Chirag018) [reply store/store.go]: yeah, missed that

**Comment by Manav-Aggarwal** [comment]
Looking at this again, this PR does not remove usage of the `multierr` package at all. It actually reverts the changes done by the issue https://github.com/rollkit/rollkit/issues/1464 and https://github.com/rollkit/rollkit/pull/1465/files#diff-9a62b4a044185786fb1bf1c5a2ea3d0d94c2e168d1ed580db8994c06a4f0f41aL78-L81. 

Please feel free to open another PR that actually removes the the usage of the `multierr` packages in different places in the codebase.

**Comment by Chirag018** [comment]
OK, my bad, instead of removing globally, I just removed it from specific file

**Pull request closed without merging**